DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Rules/AnonymizerRuleFactory.cs

```diff
@@ -0,0 +1,177 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Rules
+{
+    public class AnonymizerRuleFactory : IAnonymizerRuleFactory
+    {
+        private readonly AnonymizerDefaultSettings _defaultSettings;
+
+        private readonly Dictionary<string, JObject> _customizedSettings;
+
+        private readonly IAnonymizerProcessorFactory _processorFactory;
+
+        private readonly IAnonymizerSettingsFactory _settingsFactory;
+
+        public AnonymizerRuleFactory(AnonymizerConfiguration configuration, IAnonymizerProcessorFactory processorFactory = null, IAnonymizerSettingsFactory settingsFactory = null)
+        {
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            _defaultSettings = configuration.DefaultSettings;
+            _customizedSettings = configuration.CustomizedSettings;
+            _processorFactory = processorFactory;
+            _settingsFactory = settingsFactory;
+        }
+
+        public AnonymizerRule[] CreateAnonymizationDicomRule(JObject[] ruleContents)
+        {
+            return ruleContents?.Select(entry => CreateAnonymizationDicomRule(entry)).ToArray();
+        }
+
+        public AnonymizerRule CreateAnonymizationDicomRule(JObject ruleContent)
+        {
+            EnsureArg.IsNotNull(ruleContent, nameof(ruleContent));
+
+            // Parse and validate method
+            if (!ruleContent.ContainsKey(Constants.MethodKey))
+            {
+                throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.MissingConfigurationFields, "Missing method in rule config");
+            }
+
+            var method = ruleContent[Constants.MethodKey].ToString();
+            var supportedMethods = Enum.GetNames(typeof(AnonymizerMethod)).ToHashSet(StringComparer.InvariantCultureIgnoreCase);
```

Comment:
>Enum.GetNames [](http://example.com/codeflow?start=35&length=13)

actually this method should be in the class AnonymizerMethod, and provide a member function which returns true or false to check method is in the list or not.

Reply:
and when you put here, you will run this code for every entry, and it will affect your performance.

Reply:
Make it a static variable in this class to avoid run this every time.